addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/handler/ChamberlainMyQHandler.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.chamberlainmyq.handler;
+
+import static org.openhab.binding.chamberlainmyq.ChamberlainMyQBindingConstants.MYQ_ID;
+
+import java.io.IOException;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.openhab.binding.chamberlainmyq.config.ChamberlainMyQDeviceConfig;
+import org.openhab.binding.chamberlainmyq.handler.ChamberlainMyQGatewayHandler.RequestCallback;
+import org.openhab.binding.chamberlainmyq.internal.json.Device;
+import org.openhab.binding.chamberlainmyq.internal.json.MyqJson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ChamberlainMyQHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Hanson - Initial contribution
+ */
+public abstract class ChamberlainMyQHandler extends BaseThingHandler {
+    /**
+     * Base configuration of this device.
+     */
+    protected ChamberlainMyQDeviceConfig deviceConfig;
+
+    private Logger logger = LoggerFactory.getLogger(ChamberlainMyQHandler.class);
+
+    /**
+     * Creates a new instance of this class for the {@link Thing}.
+     *
+     * @param thing the thing that should be handled, not null.
+     */
+    public ChamberlainMyQHandler(Thing thing) {
+        super(thing);
+
+        String id = getThing().getProperties().get(MYQ_ID);
+        logger.debug("Thing ID: {}", id);
+        this.deviceConfig = new ChamberlainMyQDeviceConfig(getThing().getProperties());
```

Comment:
This is not the proper way to load the configuration, please look into `getConfigAs`